src/Microsoft.AspNet.WebUtilities/FileBufferingReadStream.cs

```diff
@@ -88,6 +101,13 @@ public override long Seek(long offset, SeekOrigin origin)
 
         private Stream CreateTempFile()
         {
+            if (_tempFileDirectory == null)
```

Comment:
Yes, I added a new constructor to this class which takes a `Func<string>`, in that case `_tempFileDirectory` will be null.